From d39e2ad6bce4706aa65fbe746ff851c494894f23 Mon Sep 17 00:00:00 2001
From: Vladimir Ivakin <vladimir_iva@pisem.net>
Date: Sun, 15 Mar 2015 01:22:37 +0300
Subject: [PATCH] add-sys-types-h-to-extract-c

---
 fscklog/extract.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/fscklog/extract.c b/fscklog/extract.c
index 318eb95..e51f4cb 100644
--- a/fscklog/extract.c
+++ b/fscklog/extract.c
@@ -28,6 +28,7 @@
 #include <stdio.h>
 #include <string.h>
 #include <unistd.h>
+#include <sys/types.h>
 
 #include "devices.h"
 #include "diskmap.h"
-- 
2.2.1

